File: src/main/java/carelog/carelog/journal/app/JournalTemplateService.java
package carelog.carelog.journal.app;

import carelog.carelog.auth.app.CustomUserDetails;
import carelog.carelog.journal.web.dto.JournalTemplateCreateRequest;
import carelog.carelog.journal.web.dto.JournalTemplateResponse;

import java.util.List;

public interface JournalTemplateService {

    JournalTemplateResponse createTemplate(JournalTemplateCreateRequest request, CustomUserDetails userDetails);
    List<JournalTemplateResponse> findAllTemplates();
}

File: src/main/kotlin/carelog/carelog/journal/app/JournalTemplateService.kt
package carelog.carelog.journal.app

import carelog.carelog.auth.app.CustomUserDetails
import carelog.carelog.journal.web.dto.JournalTemplateCreateRequest
import carelog.carelog.journal.web.dto.JournalTemplateResponse

interface JournalTemplateService {
    fun createTemplate(request: JournalTemplateCreateRequest, userDetails: CustomUserDetails): JournalTemplateResponse
    fun findAllTemplates(): List<JournalTemplateResponse>
}
